core/src/mledb/mledb-match/mledb-match.service.ts

```diff
@@ -156,4 +168,171 @@ export class MledbMatchService {
+     * all subsequent replays where those player's played need to be recalculated.
+     * @param replayId The replay to mark NCP=true/false.
+     * @param isNcp Whether the given replayId should be marked NCP or un-NCP.
+     * @returns A string containing status of what was updated.
+     */
+    async markReplaysNcp(replayIds: number[], isNcp: boolean, winningTeam?: MLE_Team): Promise<string> {
+        const r = Math.floor(Math.random() * 10000);
+        this.logger.debug(`(${r}) begin markReplaysNcp: replayIds=${replayIds}, isNcp=${isNcp}, winningTeam=${winningTeam}`);
+
+        // Winning team must be specified if NCPing replays
+        if (isNcp && !winningTeam) return "Can't mark an NCP without a winning team";
+
+        // Make sure we are considering replayIds in chronological order
+        replayIds.sort((a, b) => a - b);
+
+        // Gather replays
+        const replayPromises = replayIds.map(async rId => this.seriesReplayRepo.findOneOrFail({
+            where: {
+                id: rId,
+            },
+            relations: ["teamCoreStats.teamName"],
+        }));
+        const replays = await Promise.all(replayPromises);
+
+        // Check to make sure the winning team played in each replay
+        if (winningTeam) {
+            for (const replay of replays) {
+                const teamsInReplay = replay?.teamCoreStats.map(tcs => tcs.teamName);
+                if (!teamsInReplay.includes(winningTeam.name)) {
+                    this.logger.warn(`The team \`${winningTeam.name}\` did not play in replay with id \`${replay.id}\` (${teamsInReplay.map(t => t).join(" v. ")}), and therefore cannot be marked as the winner of this NCP. Cancelling process with no action taken.`);
+                    this.logger.warn(`Could not find team=${winningTeam.name} on replay with id=${replay.id}, cannot mark as NCP`);
+                    throw new Error(`Could not find team=${winningTeam.name} on replay with id=${replay.id}, cannot mark as NCP`);
+                }
+            }
+        }
+
+        // Set replays to NCP true/false and update winning team/color
+        for (const replay of replays) {
+            let newWinningTeam: string | undefined;
+            let newWinningColor: string | undefined;
+            
+            // Handle NCPing/Un-NCPing with non-dummy/dummy replays
+            if (isNcp) {
+                // Set winningTeam to newWinningTeam
+                if (!replay.isDummy) {
+                    // Set winningColor depending on previous/new winningTeam
+                    newWinningColor = winningTeam?.name === replay.winningTeamName
+                        ? replay.winningColor as (string | undefined)
+                        : "";
+                } else {
+                    // Set winningColor to null
+                    newWinningColor = "";
```

Comment:
"" != null?

Reply:
Comment is wrong.

Reply:
Should be null, since that's what the DB expects. Changed.